registry/remote/repository.go

```diff
@@ -1190,3 +1336,24 @@ func verifyContentDigest(resp *http.Response, expected digest.Digest) error {
 
 	return nil
 }
+
+// generateReferrersIndex generates an image index containing the given
+// referrers list in its manifests field.
+func generateReferrersIndex(referrers []ocispec.Descriptor) (ocispec.Descriptor, []byte, error) {
+	if referrers == nil {
+		referrers = []ocispec.Descriptor{} // make it an empty array to prevent potential server-side bugs
+	}
```

Comment:
If `len(referrers) == 0`, we don't even need to generate the index. Just let it go 🌬️🧊

Reply:
This is just a helper function, will handler the `len(referrers) == 0` logic outside of this function.